pkg/controller/podautoscaler/metrics/utils.go

```diff
@@ -114,10 +115,15 @@ func GetMetricsFromPods(ctx context.Context, fetcher MetricFetcher, pods []corev
 		// TODO: Let's optimize the performance for multi-metrics later.
 		metric, err := fetcher.FetchPodMetrics(ctx, pod, source)
 		if err != nil {
-			return nil, err
+			klog.ErrorS(err, "Failed to get metrics from pod", "pod", klog.KObj(&pod))
```

Comment:
There was error logging before, but if there is a persistant error then it floods the logs making them difficult to read.

I will suggest to either add metrics to track this or open to other alternative which does not floods the log.